Octokit.Tests/Clients/RepositoryHooksClientTest.cs

```diff
@@ -0,0 +1,200 @@
+﻿using NSubstitute;
+using Octokit.Tests.Helpers;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Octokit.Tests.Clients
+{
+    public class RepositoryHooksClientTests
+    {
+        public class TheGetAllMethod
+        {
+            [Fact]
+            public void RequestsCorrectUrl()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new RepositoriesClient(connection);
+
+                client.Hooks.GetAll("fake", "repo");
+
+                connection.Received().GetAll<RepositoryHook>(Arg.Is<Uri>(u => u.ToString() == "repos/fake/repo/hooks"));
+            }
+
+            [Fact]
+            public async Task EnsuresNonNullArguments()
+            {
+                var client = new RepositoriesClient(Substitute.For<IApiConnection>());
+
+                await AssertEx.Throws<ArgumentNullException>(async () => await client.Hooks.GetAll(null, "name"));
```

Comment:
Swap these out in the new tests for the `Assert.ThrowsAsync<>` that ship with xUnit 2.0 - it'll mean the inner `async/await` goes away as well...